compose/ui/ui-test/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skiko.kt

```diff
@@ -205,30 +228,50 @@ open class SkikoComposeUiTest @InternalTestApi constructor(
     private val testOwner = SkikoTestOwner()
     private val testContext = TestContext(testOwner)
 
-    fun <R> runTest(block: SkikoComposeUiTest.() -> R): R {
-        return composeRootRegistry.withRegistry {
-            withScene {
+    fun runTest(
+        block: suspend SkikoComposeUiTest.() -> Unit
+    ): TestResult {
+        composeRootRegistry.setupRegistry()
+        scene = runOnUiThread(::createUi)
+
+        @OptIn(ExperimentalStdlibApi::class)
+        val testDispatcher = runTestContext[CoroutineDispatcher] as? TestDispatcher
```

Comment:
Silently ignoring the user-provided dispatcher seems dangerous. Is this what happens in Android?
Maybe we should throw an exception if the dispatcher is not `null` and is not a `TestDispatcher`?

Reply:
Yes, that's how it's done for [Android](https://github.com/JetBrains/compose-multiplatform-core/blob/1363a6ce657384007a4d7baf5a64e70caffaf6f6/compose/ui/ui-test/src/androidMain/kotlin/androidx/compose/ui/test/ComposeUiTest.android.kt#L476) now. I'll change it. 

> Maybe we should throw an exception if the dispatcher is not null and is not a TestDispatcher?

`kotlinx.coroutines.test.runTest` will throw an exception anyway. So it's actually not us who cares about a specific type of a Dispatcher, but the underlying library (at least for now). 
I'll let the library to throw an exception. 

So it'll look like this:
```kotlin
val testDispatcher: CoroutineDispatcher = runTestContext[CoroutineDispatcher] ?: StandardTestDispatcher()
```